torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -295,142 +295,191 @@ def pipeline_llama_tracer(
     return (stage, model)
 
 
-def parallelize_llama(model, world_mesh, parallel_dims, job_config: JobConfig):
+def apply_tp(model, world_mesh, parallel_dims, job_config: JobConfig):
     """
-    Apply SPMD parallelisms and activation checkpointing to the model.
-
-    NOTE: The passed-in model preferably should be on meta device. Otherwise,
-    the model must fit on GPU or CPU memory.
+    Apply tensor parallelism.
     """
 
-    if parallel_dims.tp_enabled:
-        if job_config.model.norm_type == "fused_rmsnorm":
-            raise NotImplementedError(
-                "fused_rmsnorm not yet compatible with TP. Please use layernorm or rmsnorm."
-            )
+    assert parallel_dims.tp_enabled, "TP is not enabled"
```

Comment:
IMO no need to duplicately assert here, as we only call this function when it's enabled.

Reply:
> as we only call this function when it's enabled

Well, the function itself does not know that. So this is more of a safety measure. 
But I agree that this check can be "softer" if we just return here.

Reply:
Similar to my comment below. Parallel_dims is something shouldn't be visible to this `apply_tp` function. A function doesn't need to do assertion on whether or not itself should be called. Usually we do assertions in a function because without those assertions, the function itself cannot be applied (will throw exceptions, be divided by zero, abort abnormally, behave randomly). It is not the case here -- we can still apply the function as long as a valid mesh is given.